### DISCUSSION TITLE
Disable cache for a specific query

### DISCUSSION THREAD

**Discussion by nanandn** [TanStack/query#3021] [Q&A]
Is there a way to disable caching for a specific query? My use case is... I am fetching metric data from the server with specific intervals (1hr, 24hr, 48hr). Every time I change the interval between the three values, I want to fetch latest metrics. Since the query key is always one of the three, all of them seems to be cached and retrieved from the cache. 

How do I make it re-fetch the data everytime I change the interval?

Thanks
Anand

**Comment by TkDodo** [comment]
With the default settings, it gives you the value from the cache instantly, but then do a background refetch to give you the latest data. If you don’t want the stale data, set cacheTime to zero
